### PR TITLE
fix: improve flushing events on page leave

### PR DESCRIPTION
**Pull request by saikumarrs**
## PR Description

I've improved the effectiveness of flushing the batched events when the user leaves the page.

Additionally, a few chores were also performed:
- README files were updated to the latest.
- SDK snippets updated to the latest.
- Dependencies upgraded to the latest versions.
- RudderStack's standard favicon was added to the HTML files.
- Fixtures in the sanity suite were updated to the latest.

## Linear task (optional)

https://linear.app/rudderstack/issue/SDK-1975/sdk-does-not-effectively-track-page-close-actions-to-deliver-events-js

## Cross Browser Tests

Please confirm you have tested for the following browsers:

- [x] Chrome
- [ ] Firefox
- [x] IE11

## Sanity Suite

- [x] All sanity suite test cases pass locally

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

## Summary by CodeRabbit

- **New Features**
  - Updated RudderSnippetVersion to "3.0.10" across multiple sample apps and SDKs.
  - Added favicon links in relevant HTML files.

- **Bug Fixes**
  - Corrected import paths across various modules for consistent and error-free imports.

- **Documentation**
  - Revised README files to use "NPM" instead of "npm" for consistency.
  - Improved clarity and structure in installation and usage instructions.
  - Enhanced details in README for Chrome Extensions and Serverless Runtime usage.

- **Chores**
  - Updated `.gitignore` and `.prettierignore` to exclude new directories.

### CODE REVIEW

File: packages/analytics-js/public/index.html [hunk 1]
```diff
@@ -1,6 +1,7 @@
 <!DOCTYPE html>
 <html>
   <head>
+    <link rel="icon" href="https://rudderstack.com/favicon.ico" type="image/x-icon" sizes="48x48">
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-js/public/index.html:4]: Resolved.

File: packages/analytics-js/public/index.html [hunk 2]
```diff
@@ -83,18 +84,17 @@
               configUrl: '__CONFIG_SERVER_HOST__',
               destSDKBaseURL:
             '__DEST_SDK_BASE_URL__' + window.rudderAnalyticsBuildType + '/js-integrations',
-              pluginsSDKBaseURL: '__PLUGINS_BASE_URL__' + window.rudderAnalyticsBuildType + '/plugins',
+              pluginsSDKBaseURL: 'http://localhost:3002/cdn/' + window.rudderAnalyticsBuildType + '/plugins',
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-js/public/index.html]: Resolved.

File: packages/analytics-js-common/src/utilities/page.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+  // Catches visibility changes, such as switching tabs or minimizing the browser.
+  document.addEventListener('visibilitychange', () => {
+    if (document.visibilityState === 'hidden') {
+      handleOnLeave();
+    } else {
+      pageLeft = false;
+    }
+  });
+};
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-js-common/src/utilities/page.ts:34]: This is taken care of in another place. So, we can ignore it.

File: patches/@digitalroute+cz-conventional-changelog-for-jira+8.0.1.patch [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+ var chalk = require('chalk');
+ const { execSync } = require('child_process');
+ const boxen = require('boxen');
+@@ -72,7 +71,7 @@ module.exports = function(options) {
+   var length = longest(Object.keys(types)).length + 1;
+   var choices = map(types, function(type, key) {
+     return {
+-      name: rightPad(key + ':', length) + ' ' + type.description,
++      name: String.prototype.rightPad(key + ':', length) + ' ' + type.description,
+       value: key
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply patches/@digitalroute+cz-conventional-changelog-for-jira+8.0.1.patch]: Thanks. Done.

File: README.md [hunk 5]
```diff
@@ -39,61 +39,67 @@ The JavaScript SDK lets you track customer event data from your website and send
 - [**Usage in Chrome Extensions**](#usage-in-chrome-extensions)
 - [**Usage in Serverless Runtimes**](#usage-in-serverless-runtimes)
 
-| **IMPORTANT**: The service worker export has been deprecated from the RudderStack JavaScript SDK npm package and moved to a new package. <br/>If you still wish to use it for your project, see [**@rudderstack/analytics-js-service-worker package**](https://www.npmjs.com/package/@rudderstack/analytics-js-service-worker). |
+| **IMPORTANT**: The service worker export has been deprecated from the RudderStack JavaScript SDK NPM package and moved to a new package. <br/>If you still wish to use it for your project, see [**@rudderstack/analytics-js-service-worker package**](https://www.npmjs.com/package/@rudderstack/analytics-js-service-worker). |
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply README.md]: It is clarified in the next line.

File: packages/sanity-suite/public/v3/manualLoadCall/index-npm.html [hunk 6]
```diff
@@ -26,7 +27,7 @@
     <script>
       (function() {
         "use strict";
-        window.RudderSnippetVersion = "3.0.9";
+        window.RudderSnippetVersion = "3.0.10";
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/sanity-suite/public/v3/manualLoadCall/index-npm.html]: Not relevant.

File: packages/analytics-js/__tests__/nativeSdkLoader.js [hunk 7]
```diff
@@ -1,51 +1,65 @@
...
+          'track',
+          'identify',
+          'alias',
+          'group',
+          'reset',
+          'setAnonymousId',
+          'startSession',
+          'endSession',
+          'consent',
+        ];
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-js/__tests__/nativeSdkLoader.js]: Ignore.

File: packages/analytics-js/__tests__/nativeSdkLoader.js [hunk 8]
```diff
@@ -1,51 +1,65 @@
...
+            return function () {
+              if (Array.isArray(window[identifier])) {
+                rudderanalytics.push([methodName].concat(Array.prototype.slice.call(arguments)));
+              } else {
+                var _methodName;
+                (_methodName = window[identifier][methodName]) === null ||
+                  _methodName === void 0 ||
+                  _methodName.apply(window[identifier], arguments);
+              }
+            };
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-js/__tests__/nativeSdkLoader.js]: Ignore.

File: packages/analytics-js/__tests__/nativeSdkLoader.js [hunk 9]
```diff
@@ -1,51 +1,65 @@
...
+          'startSession',
+          'endSession',
+          'consent',
+        ];
+        for (var i = 0; i < methods.length; i++) {
+          var method = methods[i];
+          rudderanalytics[method] = (function (methodName) {
+            return function () {
+              if (Array.isArray(window[identifier])) {
+                rudderanalytics.push([methodName].concat(Array.prototype.slice.call(arguments)));
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-js/__tests__/nativeSdkLoader.js]: Ignore.

File: packages/analytics-js/__tests__/nativeSdkLoader.js [hunk 10]
```diff
@@ -1,51 +1,65 @@
 function loadingSnippet() {
   (function () {
     'use strict';
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-js/__tests__/nativeSdkLoader.js]: Ignore.

File: packages/analytics-js/__tests__/nativeSdkLoader.js [hunk 11]
```diff
@@ -1,51 +1,65 @@
...
+        window.rudderAnalyticsMount = function () {
+          if (typeof globalThis === 'undefined') {
+            Object.defineProperty(Object.prototype, '__globalThis_magic__', {
+              get: function get() {
+                return this;
+              },
+              configurable: true,
+            });
+            __globalThis_magic__.globalThis = __globalThis_magic__;
+            delete Object.prototype.__globalThis_magic__;
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-js/__tests__/nativeSdkLoader.js]: Ignore.

File: packages/analytics-js/README.md [hunk 12]
```diff
@@ -61,38 +60,33 @@ import { RudderAnalytics } from '@rudderstack/analytics-js/bundled';
...
-For examples and specific details look into [Chrome Extensions Usage](https://github.com/rudderlabs/rudder-sdk-js/blob/main/examples/chrome-extension/USAGE.md)
+For examples and specific details look into [Chrome Extensions Usage](https://github.com/rudderlabs/rudder-sdk-js/blob/main/examples/chrome-extension/USAGE.md).
 
 ## Usage in Serverless Runtimes
 
-RudderStack JS SDK [service worker](https://www.npmjs.com/package/@rudderstack/analytics-js-service-worker) can be used
-in serverless runtimes like Cloudflare Workers or Vercel Edge functions.
+RudderStack JS SDK [service worker](https://www.npmjs.com/package/@rudderstack/analytics-js-service-worker) can be used in serverless runtimes like Cloudflare Workers or Vercel Edge functions.
 
 For examples and specific details look into:
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-js/README.md]: Done.

File: packages/analytics-js/README.md [hunk 13]
```diff
@@ -61,38 +60,33 @@ import { RudderAnalytics } from '@rudderstack/analytics-js/bundled';
...
+- `npm run build:npm`: This outputs **dist/npm** folder that contains the NPM package contents.
 
 ## Usage in Chrome Extensions
 
-RudderStack JS SDK can be used in Chrome Extensions with manifest v3, both as a content script (via the JavaScript SDK package)
-or as a background script service worker (via the [service worker package](https://www.npmjs.com/package/@rudderstack/analytics-js-service-worker)).
+RudderStack JS SDK can be used in Chrome Extensions with manifest v3, both as a content script (via the JavaScript SDK package) or as a background script service worker (via the [service worker package](https://www.npmjs.com/package/@rudderstack/analytics-js-service-worker)).
 
-For examples and specific details look into [Chrome Extensions Usage](https://github.com/rudderlabs/rudder-sdk-js/blob/main/examples/chrome-extension/USAGE.md)
+For examples and specific details look into [Chrome Extensions Usage](https://github.com/rudderlabs/rudder-sdk-js/blob/main/examples/chrome-extension/USAGE.md).
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-js/README.md]: Done.

File: packages/analytics-js-plugins/README.md [hunk 14]
```diff
@@ -21,37 +21,22 @@
...
 ## How to build the plugins
 
-- Look for run scripts in the `package.json` file for getting the browser minified and non-minified builds. The builds
-- are updated in the `dist` folder of the directory. Among the others, some of the important ones are:
+- Look for run scripts in the `package.json` file for getting the browser minified and non-minified builds. The builds are updated in the `dist` folder of the directory. Among the others, some of the important ones are:
 
-  - `npm run build:browser:modern`: This outputs **dist/cdn** folder that contains the cdn package contents.
-  - `npm run build:npm`: This outputs **dist/npm** folder that contains the npm package contents.
+  - `npm run build:browser:modern`: This outputs **dist/cdn** folder that contains the CDN package contents.
+  - `npm run build:npm`: This outputs **dist/npm** folder that contains the NPM package contents.
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-js-plugins/README.md]: Done.

File: packages/analytics-js-plugins/README.md [hunk 15]
```diff
@@ -21,37 +21,22 @@
...
-packaging or as separate bundle chunks for dynamic imports.
+RudderStack JavaScript SDK plugins code that is used within the related analytics package, bundles for the legacy packaging or as separate bundle chunks for dynamic imports.
 
 > For detailed documentation on the RudderStack JavaScript SDK, click [**here**](https://www.rudderstack.com/docs/sources/event-streams/sdks/rudderstack-javascript-sdk/).
 
 ## Table of Contents
 
-- [**Installing the package**](#installing-the-package)
 - [**List of plugins**](#list-of-plugins)
 - [**How to build the plugins**](#How-to-build-the-plugins)
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-js-plugins/README.md]: Done.

File: packages/analytics-js-service-worker/README.md [hunk 16]
```diff
@@ -71,4 +69,4 @@ For examples and specific details look into:
 
 - Look for run scripts in the `package.json` file for getting the browser minified and non-minified builds. The builds are updated in the `dist` folder of the directory. Among the others, some of the important ones are:
 
-  - `npm run build:package`: This outputs **dist/npm** folder that contains the npm package contents.
+  - `npm run build:package`: This outputs **dist/npm** folder that contains the NPM package contents.
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-js-service-worker/README.md]: Done.

File: packages/analytics-js-plugins/README.md [hunk 17]
```diff
@@ -21,37 +21,22 @@
...
-```
-
-**Note that this NPM module is only meant to be used for a browser installation**. If you want to integrate RudderStack
-with your Node.js application, refer to the [**RudderStack Node.js repository**](https://github.com/rudderlabs/rudder-sdk-node).
-
 ## List of plugins
 
-Plugins are JavaScript SDK v3 features that you can optionally load on demand. Full list of available plugins can be
-explored [here](https://www.rudderstack.com/docs/sources/event-streams/sdks/rudderstack-javascript-sdk/v3/#plugins).
+Plugins are JavaScript SDK v3 features that you can optionally load on demand. Full list of available plugins can be explored [here](https://www.rudderstack.com/docs/sources/event-streams/sdks/rudderstack-javascript-sdk/load-js-sdk/#plugins).
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-js-plugins/README.md]: Done.

File: packages/analytics-js-service-worker/README.md [hunk 18]
```diff
@@ -44,23 +44,21 @@ npm install @rudderstack/analytics-js-service-worker --save
...
 | :--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
 
 ## Usage in Chrome Extensions
 
-RudderStack JS SDK can be used in Chrome Extensions with manifest v3, both as a content script (via the JavaScript SDK package)
-or as a background script service worker (via the [service worker package](https://www.npmjs.com/package/@rudderstack/analytics-js-service-worker)).
+RudderStack JS SDK can be used in Chrome Extensions with manifest v3, both as a content script (via the JavaScript SDK package) or as a background script service worker (via the [service worker package](https://www.npmjs.com/package/@rudderstack/analytics-js-service-worker)).
 
-For examples and specific details look into [Chrome Extensions Usage](https://github.com/rudderlabs/rudder-sdk-js/blob/main/examples/chrome-extension/USAGE.md)
+For examples and specific details look into [Chrome Extensions Usage](https://github.com/rudderlabs/rudder-sdk-js/blob/main/examples/chrome-extension/USAGE.md).
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-js-service-worker/README.md]: Done.

File: packages/analytics-js-service-worker/README.md [hunk 19]
```diff
@@ -22,7 +22,7 @@
 # @rudderstack/analytics-js-service-worker
 
 RudderStack JavaScript SDK service worker that can be used in browser extensions and serverless runtimes.
-It exposes the same interface and features as the [NodeJS SDK](https://www.rudderstack.com/docs/sources/event-streams/sdks/rudderstack-node-sdk/)
+It exposes the same interface and features as the [NodeJS SDK](https://www.rudderstack.com/docs/sources/event-streams/sdks/rudderstack-node-sdk/).
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-js-service-worker/README.md]: Done.

File: README.md [hunk 20]
```diff
@@ -279,33 +260,27 @@ rudderanalytics.load(<WRITE_KEY>, <DATA_PLANE_URL>, {
...
 
 - For adding or removing integrations, modify the imports in `index.js` under the `src/integrations` folder.
 
 ## Usage in Chrome extensions
 
-You can use the JavaScript SDK in Chrome Extensions with manifest v3, both as a content script (via the JavaScript SDK package)
-or as a background script service worker (via the [service worker package](https://www.npmjs.com/package/@rudderstack/analytics-js-service-worker)).
+You can use the JavaScript SDK in Chrome Extensions with manifest v3, both as a content script (via the JavaScript SDK package) or as a background script service worker (via the [service worker package](https://www.npmjs.com/package/@rudderstack/analytics-js-service-worker)).
 
 For examples and specific details, see [Chrome Extensions Usage](https://github.com/rudderlabs/rudder-sdk-js/blob/main/examples/chrome-extension/USAGE.md).
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply README.md]: Done.

File: packages/sanity-suite/README.md [hunk 21]
```diff
@@ -31,7 +31,10 @@ RudderStack JavaScript SDK sanity suite for manual and automated testing via our
...
-- https://cdn.rudderlabs.com/sanity-suite/npm/v1.1/index.html
-- https://cdn.rudderlabs.com/sanity-suite/cdn/v3/index.html
-- https://cdn.rudderlabs.com/sanity-suite/npm/v3/index.html
+- https://cdn.rudderlabs.com/sanity-suite/v3/cdn/index.html
+- https://cdn.rudderlabs.com/sanity-suite/v3/npm/index.html
+
+### Legacy SDK
+
+- https://cdn.rudderlabs.com/sanity-suite/v1.1/npm/index.html
+- https://cdn.rudderlabs.com/sanity-suite/v1.1/cdn/index.html
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/sanity-suite/README.md]: Done.

File: packages/analytics-js-service-worker/README.md (hunk 19 above)
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-js-service-worker/README.md]: Done.

File: packages/analytics-js-service-worker/README.md [hunk 22]
```diff
@@ -71,4 +69,4 @@
 
 - Look for run scripts in the `package.json` file for getting the browser minified and non-minified builds. The builds are updated in the `dist` folder of the directory. Among the others, some of the important ones are:
 
-  - `npm run build:package`: This outputs **dist/npm** folder that contains the npm package contents.
+  - `npm run build:package`: This outputs **dist/npm** folder that contains the NPM package contents.
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-js-service-worker/README.md]: Done.

File: packages/analytics-js-service-worker/README.md (hunk 18 above)
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-js-service-worker/README.md]: Done.

File: packages/analytics-js-common/src/utilities/index.ts [hunk 23]
```diff
@@ -12,3 +12,4 @@
 export * as uuId from './uuId';
 export * as timezone from './timezone';
 export * as detect from './detect';
+export * as page from './page';
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-js-common/src/utilities/index.ts:15]: The page utilities already have tests. So, we can skip this.

File: packages/sanity-suite/README.md (hunk 21 above)
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/sanity-suite/README.md]: Done.

File: README.md (hunk 20 above)
[the remark this answers is not in the document]
Comment (saikumarrs) [reply README.md]: Done.

File: README.md [hunk 24]
```diff
@@ -279,33 +260,27 @@
...
 
 - Look for run scripts in the `package.json` file for getting the browser minified and non-minified builds. The builds are updated in the `dist` folder of the directory. Among the others, some of the important ones are:
 
-  - `npm run build:browser`: This outputs **dist/cdn/legacy/rudder-analytics.min.js**.
+  - `npm run build:browser`: This outputs **dist/cdn/legacy/rsa.min.js**.
   - `npm run build:npm`: This outputs **dist/npm** folder that contains the npm package contents.
   - `npm run build:integration:all`: This outputs **dist/cdn/legacy** folder that contains the integrations.
 
-> We use **rollup** to build our SDKs. The configuration for it is present in `rollup-configs` folder.
+> We use **rollup** to build our SDKs. The configuration for it is present in `rollup-configs` directory.
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply README.md]: Done.

File: README.md [hunk 25]
```diff
@@ -279,33 +260,27 @@
...
-You can start adding integrations of your choice for sending the data through their respective web (JavaScript) SDKs.
-
 ## How to build the SDK
 
 - Look for run scripts in the `package.json` file for getting the browser minified and non-minified builds. The builds are updated in the `dist` folder of the directory. Among the others, some of the important ones are:
 
-  - `npm run build:browser`: This outputs **dist/cdn/legacy/rudder-analytics.min.js**.
+  - `npm run build:browser`: This outputs **dist/cdn/legacy/rsa.min.js**.
   - `npm run build:npm`: This outputs **dist/npm** folder that contains the npm package contents.
   - `npm run build:integration:all`: This outputs **dist/cdn/legacy** folder that contains the integrations.
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply README.md]: Done.

File: README.md [hunk 26]
```diff
@@ -279,35 +260,29 @@
...
 ## How to build the SDK
 
 - Look for run scripts in the `package.json` file for getting the browser minified and non-minified builds. The builds are updated in the `dist` folder of the directory. Among the others, some of the important ones are:
 
-  - `npm run build:browser`: This outputs **dist/cdn/legacy/rudder-analytics.min.js**.
-  - `npm run build:npm`: This outputs **dist/npm** folder that contains the npm package contents.
-  - `npm run build:integration:all`: This outputs **dist/cdn/legacy** folder that contains the integrations.
+  - `npm run build:browser`: This outputs the **dist/cdn/legacy/rsa.min.js**.
+  - `npm run build:npm`: This outputs the **dist/npm** folder that contains the NPM package contents.
+  - `npm run build:integration:all`: This outputs the **dist/cdn/legacy** folder that contains the integrations.
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply README.md:269]: Done.

File: README.md [hunk 27]
```diff
@@ -39,61 +39,67 @@
...
 
 <br>
 
-> The above snippet lets you integrate the SDK with your website and load it asynchronously to keep your page load time unaffected.
+> The above snippet lets you integrate the SDK with your website and load it asynchronously to avoid impacting the performance of your webpages.
 
 To load SDK script on to your page synchronously, see the [**JavaScript SDK documentation**](https://www.rudderstack.com/docs/sources/event-streams/sdks/rudderstack-javascript-sdk/installation/#synchronous-loading).
 
-> **IMPORTANT**: The implicit `page` call at the end of the snippet (present in the previous JavaScript SDK versions) is removed in the latest SDK v3. You need to make a `page` call explicitly, if required, as shown below:
+> **IMPORTANT**: The implicit `page` call at the end of the snippet (present in the previous JavaScript SDK versions) is removed in the latest SDK (v3). You need to make a `page` call explicitly, if required, as shown below:
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply README.md:86]: It is well documented. Ignore it.

File: README.md [hunk 28]
```diff
@@ -39,61 +39,67 @@ The JavaScript SDK lets you track customer event data from your website and send
...
-  document.head?document.head.appendChild(e):document.body.appendChild(e)
-  },"undefined"==typeof Promise||"undefined"==typeof globalThis){var d=document.createElement("script")
-  ;d.src="https://polyfill-fastly.io/v3/polyfill.min.js?version=3.111.0&features=Symbol%2CPromise&callback=rudderAnalyticsMount",
-  d.async=asyncScript,document.head?document.head.appendChild(d):document.body.appendChild(d)}else{
-  window.rudderAnalyticsMount()}window.rudderanalytics.load(<WRITE_KEY>,<DATA_PLANE_URL>,{})}();
+  delete Object.prototype.__globalThis_magic__),
+  window.rudderAnalyticsAddScript("".concat(sdkBaseUrl,"/").concat(window.rudderAnalyticsBuildType,"/").concat(sdkName),"data-rsa-write-key",<WRITE_KEY>)
+  },
+  "undefined"==typeof Promise||"undefined"==typeof globalThis?window.rudderAnalyticsAddScript("https://polyfill-fastly.io/v3/polyfill.min.js?version=3.111.0&features=Symbol%2CPromise&callback=rudderAnalyticsMount"):window.rudderAnalyticsMount()
+  ;t.load(<WRITE_KEY>,<DATA_PLANE_URL>,{})}}}();
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply README.md:76]: No need as it'll be evident on the browser console logs.

File: README.md [hunk 29]
```diff
@@ -279,35 +260,29 @@
...
+For more details, see [Chrome Extensions Usage](https://github.com/rudderlabs/rudder-sdk-js/blob/main/examples/chrome-extension/USAGE.md).
 
 ## Usage in Serverless runtimes
 
-RudderStack JS SDK [service worker](https://www.npmjs.com/package/@rudderstack/analytics-js-service-worker) can be used
-in serverless runtimes like Cloudflare Workers or Vercel Edge functions.
+RudderStack JS SDK [service worker](https://www.npmjs.com/package/@rudderstack/analytics-js-service-worker) can be used in serverless runtimes like Cloudflare Workers or Vercel Edge functions.
 
-For examples and specific details look into:
+For more details see:
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply README.md]: Done.

File: README.md (hunk 28 above)
[the remark this answers is not in the document]
Comment (saikumarrs) [reply README.md:76]: Not required.

**Review by MoumitaM — DISMISSED** [review]

File: packages/analytics-js/src/services/StoreManager/storages/storageEngine.ts [hunk 30]
```diff
@@ -12,7 +12,7 @@ import {
   SESSION_STORAGE,
 } from '@rudderstack/analytics-js-common/constants/storages';
 import type { StorageType } from '@rudderstack/analytics-js-common/types/Storage';
-import { state } from '@rudderstack/analytics-js/state';
+import { state } from '../../../state';
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-js/src/services/StoreManager/storages/storageEngine.ts:15]: It is fine as CookieStorage implements a global singleton.

**Pull request merged**